relayer/provider/cosmos/query.go

```diff
@@ -59,7 +72,32 @@ func (cc *CosmosProvider) QueryTxs(ctx context.Context, page, limit int, events
 	if err != nil {
 		return nil, err
 	}
-	return res.Txs, nil
+
+	var txResps []*provider.RelayerTxResponse
```

Comment:
`txResps` can be pre-sized to match `res.Txs`. Pre-sizing should eliminate how often the underlying slice needs to grow, which reduces garbage collector pressure at near-zero cost to code readability and maintainability.